### ISSUE TITLE
PG::UndefinedTable: ERROR:  relation "application_settings" does not exist (Puma && Sidekiq)

### ISSUE THREAD

**Issue by lenkeybalis** [sameersbn/docker-gitlab#2221]
Dear all,

There is problem with my gitlab. During the initiallization I get the following error code. Basically I run these application on Kubernetes 1.18.3.

I use this with the simple configuration.  I just rewrote  from "ReplicaController"to "Deployment" kind. That's all. Everything is the same.
Components:
 - Gitlab 13.3.1
 - Postgres 11
 - Redis
 - Minio (it is plus component what I want to use for my runner pod in the future)

I observed the following problem during the initalization:

Puma log and Sidekiq:

WARNING: PID file creation will be removed in Sidekiq 6.0, see #4045. Please use a proper process supervisor to start and manage your services
WARNING: Logfile redirection will be removed in Sidekiq 6.0, see #4045. Sidekiq will only log to STDOUT
PG::UndefinedTable: ERROR:  relation "application_settings" does not exist
LINE 1: SELECT "application_settings".* FROM "application_settings" ...

When I see this error in the logs, then in parallel also I see some error rows in postresql to.

Gitlab logs:
2015-07-18 13:35:40,282 INFO exited: puma (exit status 1; not expected)
2015-07-18 13:35:41,286 INFO spawned: 'puma' with pid 296
2015-07-18 13:35:41,287 INFO success: puma entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2015-07-18 13:35:41,628 INFO exited: sidekiq (exit status 1; not expected)
2015-07-18 13:35:41,286 INFO spawned: 'sidekiq' with pid 298
2015-07-18 13:35:42,629 INFO success: sidekiq entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)

Could you please help me in this issue? Thank you.

Basically I use this stack with dyn provisoning sc. (Rook), Each components are working fine and the connection is also stable from my perspective. The starnge things the issue occured a brand-new installation.

Thank for your help!

**Comment by lacisz491** [comment]
Hi,

I have the same issue right now.

**Comment by asparrowhawk** [comment]
I am having the same issue when trying to get a fresh system running based on release 17.0.2.

I created a new VM with the latest Alpine Linux (virt) as the O/S. I then installed docker and docker compose cli. Downloaded the `docker-compose.yml` file, setup the various secrets and settings, finally starting the containers.

In the logs for the `postgresql` container I see:

```
docker compose logs postgresql
postgresql-1  | Initializing datadir...
postgresql-1  | Initializing certdir...
postgresql-1  | Initializing logdir...
postgresql-1  | Initializing rundir...
postgresql-1  | Setting resolv.conf ACLs...
postgresql-1  | Initializing database...
postgresql-1  | Configuring hot standby...
postgresql-1  | ‣ Setting postgresql.conf parameter: wal_level = 'hot_standby'
postgresql-1  | ‣ Setting postgresql.conf parameter: max_wal_senders = '16'
postgresql-1  | ‣ Setting postgresql.conf parameter: checkpoint_segments = '8'
postgresql-1  | ‣ Setting postgresql.conf parameter: wal_keep_segments = '32'
postgresql-1  | ‣ Setting postgresql.conf parameter: hot_standby = 'on'
postgresql-1  | ‣ Setting postgresql.conf parameter: data_directory = '/var/lib/postgresql/14/main'
postgresql-1  | ‣ Setting postgresql.conf parameter: log_directory = '/var/log/postgresql'
postgresql-1  | ‣ Setting postgresql.conf parameter: log_filename = 'postgresql-14-main.log'
postgresql-1  | ‣ Setting postgresql.conf parameter: ssl = 'off'
postgresql-1  | Creating database user: gitlab
postgresql-1  | Creating database: gitlabhq_production...
postgresql-1  | ‣ Loading pg_trgm extension...
postgresql-1  | ‣ Loading btree_gist extension...
postgresql-1  | ‣ Granting access to gitlab user...
postgresql-1  | Starting PostgreSQL 14...
postgresql-1  | 2024-06-17 10:18:35.341 UTC [1] LOG:  starting PostgreSQL 14.8 (Ubuntu 14.8-1.pgdg22.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0, 64-bit
postgresql-1  | 2024-06-17 10:18:35.342 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgresql-1  | 2024-06-17 10:18:35.342 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgresql-1  | 2024-06-17 10:18:35.712 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgresql-1  | 2024-06-17 10:18:36.705 UTC [159] FATAL:  the database system is starting up
postgresql-1  | 2024-06-17 10:18:36.708 UTC [158] LOG:  database system was shut down at 2024-06-17 10:18:34 UTC
postgresql-1  | 2024-06-17 10:18:37.012 UTC [1] LOG:  database system is ready to accept connections
postgresql-1  | 2024-06-17 10:21:38.181 UTC [179] ERROR:  relation "feature_gates" does not exist at character 491
postgresql-1  | 2024-06-17 10:21:38.181 UTC [179] STATEMENT:  SELECT a.attname, format_type(a.atttypid, a.atttypmod),
postgresql-1  | 	       pg_get_expr(d.adbin, d.adrelid), a.attnotnull, a.atttypid, a.atttypmod,
postgresql-1  | 	       c.collname, col_description(a.attrelid, a.attnum) AS comment,
postgresql-1  | 	       attgenerated as attgenerated
postgresql-1  | 	  FROM pg_attribute a
postgresql-1  | 	  LEFT JOIN pg_attrdef d ON a.attrelid = d.adrelid AND a.attnum = d.adnum
postgresql-1  | 	  LEFT JOIN pg_type t ON a.atttypid = t.oid
postgresql-1  | 	  LEFT JOIN pg_collation c ON a.attcollation = c.oid AND a.attcollation <> t.typcollation
postgresql-1  | 	 WHERE a.attrelid = '"feature_gates"'::regclass
postgresql-1  | 	   AND a.attnum > 0 AND NOT a.attisdropped
postgresql-1  | 	 ORDER BY a.attnum
postgresql-1  |
```

Using `exec gitlab supervisorctl tail -f sidekiq` I see this:
```
PG::UndefinedTable: ERROR:  relation "application_settings" does not exist
LINE 9:  WHERE a.attrelid = '"application_settings"'::regclass
                            ^
/home/git/gitlab/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.8.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `exec'
/home/git/gitlab/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.8.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `block (2 levels) in query'
/home/git/gitlab/vendor/bundle/ruby/3.2.0/gems/activesupport-7.0.8.1/lib/active_support/concurrency/share_lock.rb:187:in `yield_shares'
/home/git/gitlab/vendor/bundle/ruby/3.2.0/gems/activesupport-7.0.8.1/lib/active_support/dependencies/interlock.rb:41:in `permit_concurrent_loads'
/home/git/gitlab/vendor/bundle/ruby/3.2.0/gems/activerecord-7.0.8.1/lib/active_record/connection_adapters/postgresql/database_statements.rb:18:in `block in query'
/home/git/gitlab/vendor/bundle/ruby/3.2.0/gems/activesupport-7.0.8.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/home/git/gitlab/vendor/bundle/ruby/3.2.0/gems/activesupport-7.0.8.1/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
...
```

So it appears that the database isn't being initialised correctly. Does anyone have any ideas as to what is wrong?

**Comment by flavienbwk** [comment]
I had this error during restoration. I just had to wait, it was working in the background.
